code/modules/client/preferences.dm

```diff
@@ -547,6 +547,9 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 				else
 					HTML += "<font color=red>[rank]</font></td><td><font color=red><b> \[NON-HUMAN\]</b></font></td></tr>"
 				continue
+			if((rank == "AI") && !(config.allow_ai))
+				HTML += "<font color=red>[rank]</font></td><td><font color=red><b> \[DISABLED\]</b></font></td></tr>"
+				continue

```

Comment:
also, this is still technically a href exploit. if preferences/process_link (or what ever its called) doesn't also check these same variables, then somebody could abuse that.

but it seems that job selection code properly handles that anyways, so its only technically a href exploit. (and its not like you can't just also do it by loading a preference that already has it selected.)